### ISSUE TITLE
Auto Encoder normalization

### ISSUE THREAD

**Issue by George3d6** [mindsdb/lightwood#206]
Currently out auto encoders (by which I mean the RNN autoencoder and the Categorical autoencoder) don't give us a guarantee that they output values within a certain range.

Ideally the autoencoders (and all encoders really) should output values within a fairly tiny range (say 90% of values between -2 and 2 and the rest with an abs value < 100... but these are just arbitrary numbers to illustrate the point).

This would ideally help the mixer train much faster, even though in theory neural nets can use unormalized inputs, in practice it seems that normalizing them helps (as opposed to e.g. having some in the 0-1 range and other in the 10^3-10^4 range)... but @paxcema @torrmal if you've got some citations for/against this point I'd be glad to hear them, since it might help us figure out what our normalization aim is.

At any rate, the alternative to normalizing in the autoencoders themselves is adding a normalization layer, I'm against that because:

1. log representations wouldn't be normalized properly if it's a linear norm (and vice versa, and generalized to a lot of functions which are incompatible with whatever normalization function we chose in terms of preserving the original meaning)
2. Depending on the actual function used there's no guarantee that the embedding layer's values can be normalized without loosing meaning (i.e. I don't think we have a guarantee the embeddings from auto encoders are linear if we'd decide to go for a linear norm)
3. Seems like something that would make connecting the encoding networks and training them together with the mixer even harder

***

Thus I think it's safe to say we want to normalize in the autoencoders themselves.

My proposed solution is to just have a "normalizing" activation function for the embedding layer. But @paxcema whenever you pick a specific solution for the RNN autoencoder you can leave the details here and we can implement it for the categorical auto encoder.

**Comment by btseytlin** [comment]
A good idea might be to just use a small l1/l2 regularization when training these encoders (`weight_decay=1e6` argument for the optimizer). This does not guarantee a range, but the embeddings will usually be somewhere around -1 and 1

**Comment by paxcema** [comment]
With respect to the time series encoder-decoder, the current embedding is the hidden state that the GRU encoder generates. As such, and given the GRU cell inner operations, we have a guarantee that each embedding component will be in the [-1, 1] range, which is what we probably want. Having said that, @btseytlin idea of weight decay is a good practice as well, so I've implemented it in the PR. 
I will check the categorical autoencoder and comment back on ideas for it.

**Comment by George3d6** [comment]
Done.
